--- configure.orig	2004-11-08 13:54:56.000000000 +0100
+++ configure	2004-11-08 13:55:18.000000000 +0100
@@ -3471,12 +3471,12 @@
 
 
 
-sed -e 's@\(HREF="[^"]*\)\.in">@\1">@' INSTALL.html > INSTALL.html$$
-rm INSTALL.html; mv INSTALL.html$$ INSTALL.html
+#sed -e 's@\(HREF="[^"]*\)\.in">@\1">@' INSTALL.html > INSTALL.html$$
+#rm INSTALL.html; mv INSTALL.html$$ INSTALL.html
 
 
-sed -e 's@\(HREF="[^"]*\)\.in">@\1">@' FAQ.html > FAQ.html$$
-rm FAQ.html; mv FAQ.html$$ FAQ.html
+#sed -e 's@\(HREF="[^"]*\)\.in">@\1">@' FAQ.html > FAQ.html$$
+#rm FAQ.html; mv FAQ.html$$ FAQ.html
 
 
 
