diff --git a/lib/Mojo/XMLRPC.pm b/lib/Mojo/XMLRPC.pm
index 5e30d5f..2ec4b6a 100644
--- a/lib/Mojo/XMLRPC.pm
+++ b/lib/Mojo/XMLRPC.pm
@@ -180,7 +180,16 @@ sub _decode_element {
     my $date = Mojo::Date->new($elem->text);
     unless ($date->epoch) {
       require Time::Piece;
-      $date->epoch(Time::Piece->strptime($elem->text, '%Y%m%dT%H:%M:%S')->epoch);
+      my $text = $elem->text;
+      PARSE: for my $time_format ('%H:%M:%S', '%H%M%S') {
+        for my $calendar_format ('%Y%m%d', '%Y-%m-%d') {
+          my $format = $calendar_format . 'T' . $time_format;
+          eval {
+            $date->epoch(Time::Piece->strptime($text, $format)->epoch);
+          };
+          last PARSE unless $@;
+        }
+      }
     }
     return $date;
   } elsif ($tag eq 'base64') {
diff --git a/t/from_xmlrpc.t b/t/from_xmlrpc.t
index 4eb7a50..caa718e 100644
--- a/t/from_xmlrpc.t
+++ b/t/from_xmlrpc.t
@@ -135,6 +135,25 @@ MESSAGE
   is $date->epoch, 900684535, 'got the correct date';
 }
 
+$msg = from_xmlrpc(<<'MESSAGE');
+<?xml version="1.0"?>
+<methodResponse>
+   <params>
+      <param>
+         <value><dateTime.iso8601>1969-12-31T235959</dateTime.iso8601></value>
+         </param>
+      </params>
+   </methodResponse>
+MESSAGE
+
+{
+  isa_ok $msg, 'Mojo::XMLRPC::Message::Response', 'correct message type';
+  ok !$msg->is_fault, 'not a fault';
+  my $date = $msg->parameters->[0];
+  isa_ok $date, 'Mojo::Date', 'got a Mojo::Date';
+  is $date->epoch, -1, 'got the correct date';
+}
+
 $msg = from_xmlrpc(<<'MESSAGE');
 <?xml version="1.0"?>
 <methodResponse>
